zest/releaser/release.py

```diff
@@ -144,35 +141,15 @@ def _upload_distributions(self, package):
             servers = self.pypiconfig.distutils_servers()
 
         for server in servers:
-            if server == 'pypi':
-                if on_pypi:
-                    logger.info("This package is registered on PyPI.")
-                    # Already registered on PyPI.  Uploading is enough.
-                    do_register = False
-                    question = "Upload"
-                else:
-                    # We must register first.
-                    do_register = True
-                    question = "Register and upload"
-            else:
-                # Assume register unnecessary unless publishing to Public PyPi.
-                do_register = False
-                question = "Upload"
+            if self.pypiconfig.register_package():
+                logger.info("Registering...")
```

Comment:
If a user has multiple servers in his `~.pypirc`, this change means we register at all three servers, without asking any questions.
I would prefer this:
- Ask if user wants to upload to server X.
- No: don't upload and don't register.
- Yes: if register is on in the config, we register. And then we upload.